binaries/daemon/src/node_communication/tcp.rs

```diff
@@ -74,9 +74,90 @@ impl Connection for TcpConnection {
+                    let variant_name = match variant_idx {
+                        0 => "Register",
+                        1 => "Subscribe",
+                        2 => "SendMessage",
+                        3 => "CloseOutputs",
+                        4 => "OutputsDone",
+                        5 => "NextEvent",
+                        6 => "ReportDropTokens",
+                        7 => "SubscribeDrop",
+                        8 => "NextFinishedDropTokens",
+                        9 => "EventStreamDropped",
+                        10 => "NodeConfig",
+                        _ => "Unknown",
+                    };
+                    format!(
+                        " (variant index: {} = {}, expected 8 for SubscribeDrop)",
+                        variant_idx, variant_name
+                    )
+                } else {
+                    String::new()
+                };
+
+                // Workaround for sync/async I/O mismatch: if we receive variant 7 (ReportDropTokens)
+                // with a 28-byte message (which matches SubscribeDrop size), it's likely a corrupted
+                // SubscribeDrop message due to sync/async I/O mismatch. Try to fix it.
+                let mut fixed_raw = raw.clone();
+                if raw.len() == 28 && raw.len() >= 4 {
+                    let variant_idx = u32::from_le_bytes([raw[0], raw[1], raw[2], raw[3]]);
+                    if variant_idx == 7 {
+                        // Change variant index from 7 (ReportDropTokens) to 8 (SubscribeDrop)
+                        tracing::debug!(
+                            "attempting to fix corrupted SubscribeDrop message (variant 7 -> 8) due to sync/async I/O mismatch"
+                        );
+                        fixed_raw[0] = 8;
+                        fixed_raw[1] = 0;
+                        fixed_raw[2] = 0;
+                        fixed_raw[3] = 0;
+                        // Try to deserialize the fixed message
+                        if let Ok(message) =
+                            bincode::deserialize::<Timestamped<DaemonRequest>>(&fixed_raw)
+                        {
+                            if matches!(message.inner, DaemonRequest::SubscribeDrop) {
+                                tracing::debug!(
+                                    "successfully fixed corrupted SubscribeDrop message"
+                                );
+                                return Ok(Some(message));
+                            }
+                        }
+                    }
+                }
+
+                tracing::warn!(
+                    "failed to deserialize DaemonRequest: {err}, raw data length: {}, preview: {}, hex: {}{}, treating as disconnect",
+                    raw.len(),
+                    preview,
+                    hex_preview,
+                    variant_hint
+                );
+                Ok(None)
+            }
+        }
     }
 
```

Comment:
In general, there should not be any deserialization errors because we use reliable channels. Could you clarify why you think that this change is needed?

Either way, this change it not really related to this PR. So let's move it to a separate one if there is good motivation to do this.